src/shapes/text.class.js

```diff
@@ -1109,6 +1106,17 @@
       if (decl && decl.textBackgroundColor) {
         this._removeShadow(ctx);
       }
+      if (decl && decl.deltaY) {
+        var height = this.getHeightOfChar(lineIndex, charIndex);
+        var deltaY = decl.deltaY;
+        if (deltaY <= 0) {
+          deltaY = height - Math.min(height - deltaY, lineHeight * (1 - LINE_PADDING));
+        } else {
+          deltaY = Math.max(deltaY, lineHeight * LINE_PADDING);
+        }
+        top += deltaY;
+      }
```

Comment:
what is this doing? why we cannot have just top += deltay? i do not understand the math purpose.